src/zarr/group.py

```diff
@@ -401,50 +465,54 @@ def attrs(self) -> Attributes:
     def info(self):
         return self._async_group.info
 
-    @property
-    def store_path(self) -> StorePath:
-        return self._async_group.store_path
-
-    def update_attributes(self, new_attributes: Dict[str, Any]):
+    def update_attributes(self, new_attributes: dict[str, Any]):
         self._sync(self._async_group.update_attributes(new_attributes))
         return self
 
     @property
-    def nchildren(self) -> int:
-        return self._sync(self._async_group.nchildren())
+    def nmembers(self) -> int:
+        return self._sync(self._async_group.nmembers())
 
     @property
-    def children(self) -> List[Union[Array, Group]]:
-        raise NotImplementedError
-        # Uncomment with AsyncGroup implements this method
-        # _children: List[Union[AsyncArray, AsyncGroup]] = self._sync_iter(
-        #     self._async_group.children()
-        # )
-        # return [Array(obj) if isinstance(obj, AsyncArray) else Group(obj) for obj in _children]
+    def members(self) -> tuple[tuple[str, Array | Group], ...]:
```

Comment:
Let's continue to think about the return type here. I could see a (frozen) dict being a more ergonomic return type.

Reply:
I'm also a fan of something immutable and dict-like here. I think we can use  #1787 to continue brainstorming